server/src/main/java/org/spine3/server/stand/QueryProcessor.java

```diff
@@ -0,0 +1,223 @@
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+package org.spine3.server.stand;
+
+import com.google.common.base.Function;
+import com.google.common.base.Predicate;
+import com.google.common.collect.Collections2;
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.ImmutableCollection;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.Any;
+import com.google.protobuf.FieldMask;
+import com.google.protobuf.Message;
+import io.grpc.stub.StreamObserver;
+import org.spine3.base.Queries;
+import org.spine3.client.EntityFilters;
+import org.spine3.client.EntityId;
+import org.spine3.client.EntityIdFilter;
+import org.spine3.client.Query;
+import org.spine3.client.Target;
+import org.spine3.protobuf.AnyPacker;
+import org.spine3.protobuf.TypeUrl;
+import org.spine3.server.entity.Entity;
+import org.spine3.server.entity.EntityRepository;
+import org.spine3.server.storage.EntityStorageRecord;
+import org.spine3.server.storage.StandStorage;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Query processing helper utility.
+ *
+ * @author Alex Tymchenko
+ */
+/* package */
+@SuppressWarnings("UtilityClass")
+class QueryProcessor {
+
+    private QueryProcessor() {
+    }
+
+    /**
+     * Performs query processing as a part of {@link Stand#execute(Query, StreamObserver)}.
+     *
+     * @param query               an instance of {@code Query} to process
+     * @param standStorage        internal storage of {@link Stand}
+     * @param aggregateTypes      {@link org.spine3.server.aggregate.Aggregate} types exposed by the {@code Stand}
+     * @param typeToRepositoryMap map of exposed types and related repositories registered in the {@code Stand}
+     * @return the query result
+     */
+    /* package */
+    static ImmutableCollection<Any> processQuery(Query query,
+                                                 StandStorage standStorage,
+                                                 Set<TypeUrl> aggregateTypes,
+                                                 Map<TypeUrl, EntityRepository<?, ? extends Entity, ? extends Message>> typeToRepositoryMap) {
```

Comment:
Reworked. Should not be an issue anymore.

This structure must belong to a `Stand`, declared inside of `Stand` only and not be propagated somewhere else.